components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/EnforcePasswordResetAuthenticationHandler.java

```diff
@@ -43,13 +43,17 @@
 import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
 
 import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
 import java.util.List;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 public class EnforcePasswordResetAuthenticationHandler extends AbstractPostAuthnHandler {
 
     private static final Log log = LogFactory.getLog(EnforcePasswordResetAuthenticationHandler.class);
+    private static final String PASSWORD_EXPIRED_MSG = URLEncoder.encode(
```

Comment:
Any particular reason to define this constant? Can't we have this inline?

Reply:
One reason is to future-proof the password expired message by making it easier to modify or reuse.

Reply:
Also, we avoid repeated calls to the encode function.